lib/coffee_place/cli.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require_relative '../coffee_place'
+require_relative './location'
+require_relative './cli_parser'
+require_relative './importer'
+require_relative './search'
+
+module CoffeePlace
+  # Parses and runs CLI application using supplied args.
+  class CLI
+    attr_accessor :cli_opts, :parsed_args
+
+    def initialize(parser: CLIParser.new,
+                   importer: Importer.new)
+      @parser = parser
+      @importer = importer
+    end
+
+    def run(args)
+      cli_opts = parse_cli_options!(args)
+
+      show_help_and_exit! if cli_opts.print_help?
+      show_version_and_exit! if cli_opts.show_version?
```

Comment:
I'm curious - why did you write up the methods with a bang?

Reply:
I like using a bang when naming methods that have important side effects.
This warns the programmer to be weary of side effects.
These methods exit the program - which is pretty severe, as far as side effects code.

In `ActiveRecord`'s design, most methods with a bang have the side effect of throwing errors.
(e.g. `save!`, `validate!`)

In Ruby's standard library, a bang usually means that a method mutates data in place, instead of creating
a new data structure - for example [Array#map!](https://ruby-doc.org/core-3.0.2/Array.html#method-i-map-21)

Reply:
Yes, I imagined that's the reason. Thank you for answering. :)